eng/common/pipelines/templates/steps/run-specific-variables.yml

```diff
@@ -0,0 +1,34 @@
+# Add Run Specific variables such as Language and release intent
+# $Language = "net", "java", "js", "python"
+# $IsReleaseRun = "true", "false"
+
+steps:
+  - pwsh: |
+      # Determine Intent of the run i.e. if it is intended for release
+      $isForRelease = $False
+      if ('$(Build.Reason)' -eq 'Manual' -and '$(System.TeamProject)' -eq 'internal')
+      {
+        $isForRelease = $True
+      }
+      # Determine Repo language from the Repo Name
+      if ('$(Build.Repository.Name)' -match "azure-sdk-for-(?<lang>[^-]+)")
```

Comment:
I like the for release variable as it parallels the dev build variable stuff (although we should allow for someone to override it like we do for the dev variable). However for the language I have a different idea in mind. In particular it should be static for each language repo so we should either create a common ps1 script we always import from our eng/common scripts. In there we can define a global language variable statically, instead of depending on the repo name.